src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_QueryLambdaRewriter.vb

```diff
@@ -0,0 +1,93 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Symbols
+
+Namespace Microsoft.CodeAnalysis.Semantics
+    Partial Friend NotInheritable Class VisualBasicOperationFactory
+        Private Shared Function RewriteQueryLambda(node As BoundQueryLambda) As BoundNode
+            ' We rewrite query lambda into regular lambda with 2 passes.
+            ' Pass 1 uses helper methods from LocalRewriter to do the lowering. This introduces large number of DAGs.
+            ' Pass 2 walks over the lowered tree and replaces duplicate bound nodes in the tree with their clones - this is a requirement for the Operation tree.
+            Dim pass1Rewriter As New QueryLambdaRewriterPass1
+            Dim rewrittenLambda As BoundLambda = DirectCast(pass1Rewriter.VisitQueryLambda(node), BoundLambda)
+
+            Dim pass2Rewriter As New QueryLambdaRewriterPass2
+            Return pass2Rewriter.VisitLambda(rewrittenLambda)
+        End Function
+
+        Private NotInheritable Class QueryLambdaRewriterPass1
+            Inherits BoundTreeRewriterWithStackGuard
+            Private _rangeVariableMap As Dictionary(Of RangeVariableSymbol, BoundExpression)
+
+            Public Sub New()
+                _rangeVariableMap = Nothing
+            End Sub
+
+            Public Overrides Function VisitQueryLambda(node As BoundQueryLambda) As BoundNode
+                LocalRewriter.PopulateRangeVariableMapForQueryLambdaRewrite(node, _rangeVariableMap, inExpressionLambda:=False)
```

Comment:
>False [](start = 121, length = 5)

We probably want to pass True here to get property access nodes instead of accessor calls.